crates/uv-trampoline/tests/harness.rs

```diff
@@ -0,0 +1,239 @@
+/// stored zip file.
+///
+/// <https://github.com/pypa/pip/blob/fd0ea6bc5e8cb95e518c23d901c26ca14db17f89/src/pip/_vendor/distlib/scripts.py#L248-L262>
+#[allow(unused_variables)]
+fn windows_script_launcher(
+    launcher_python_script: &str,
+    is_gui: bool,
+    python_executable: impl AsRef<Path>,
+) -> Result<Vec<u8>, Error> {
+    // This method should only be called on Windows, but we avoid `#[cfg(windows)]` to retain
+    // compilation on all platforms.
+    if cfg!(not(windows)) {
+        return Err(Error::NotWindows);
+    }
+
+    let launcher_bin: &[u8] = match env::consts::ARCH {
+        #[cfg(all(windows, target_arch = "x86"))]
+        "x86" => {
+            if is_gui {
+                LAUNCHER_I686_GUI
+            } else {
+                LAUNCHER_I686_CONSOLE
+            }
+        }
+        #[cfg(all(windows, target_arch = "x86_64"))]
+        "x86_64" => {
+            if is_gui {
+                LAUNCHER_X86_64_GUI
+            } else {
+                LAUNCHER_X86_64_CONSOLE
+            }
+        }
+        #[cfg(all(windows, target_arch = "aarch64"))]
+        "aarch64" => {
+            if is_gui {
+                LAUNCHER_AARCH64_GUI
+            } else {
+                LAUNCHER_AARCH64_CONSOLE
+            }
+        }
+        #[cfg(windows)]
+        arch => {
+            return Err(Error::UnsupportedWindowsArch(arch));
+        }
+        #[cfg(not(windows))]
+        arch => &[],
+    };
+
+    let mut payload: Vec<u8> = Vec::new();
+    {
+        // We're using the zip writer, but with stored compression
+        // https://github.com/njsmith/posy/blob/04927e657ca97a5e35bb2252d168125de9a3a025/src/trampolines/mod.rs#L75-L82
+        // https://github.com/pypa/distlib/blob/8ed03aab48add854f377ce392efffb79bb4d6091/PC/launcher.c#L259-L271
+        let stored = FileOptions::default().compression_method(zip::CompressionMethod::Stored);
+        let mut archive = ZipWriter::new(Cursor::new(&mut payload));
+        let error_msg = "Writing to Vec<u8> should never fail";
+        archive.start_file("__main__.py", stored).expect(error_msg);
+        archive
+            .write_all(launcher_python_script.as_bytes())
+            .expect(error_msg);
+        archive.finish().expect(error_msg);
+    }
+
+    let python = python_executable.as_ref();
+    let python_path = python.simplified_display().to_string();
+
+    let mut launcher: Vec<u8> = Vec::with_capacity(launcher_bin.len() + payload.len());
+    launcher.extend_from_slice(launcher_bin);
+    launcher.extend_from_slice(&payload);
+    launcher.extend_from_slice(python_path.as_bytes());
+    launcher.extend_from_slice(
+        &u32::try_from(python_path.as_bytes().len())
+            .expect("File Path to be smaller than 4GB")
+            .to_le_bytes(),
+    );
+    launcher.extend_from_slice(&LAUNCHER_MAGIC_NUMBER);
+
+    Ok(launcher)
+}
+
+#[test]
+fn generate_test_launchers() -> Result<()> {
```

Comment:
Could you split this into two tests? One that runs the CLI test, and another that needs the user to close the window annotated with `#[ignore]`

Reply:
Done in 20df349646b6ac85fb9b7e01098f6018e4d67f76